terraform/modules/app-ecs-albs/main.tf

```diff
@@ -279,14 +287,97 @@ resource "aws_route53_record" "alerts_private_record" {
   }
 }
 
+######################################################################
+# ----- prometheus public ALB -------
+######################################################################
+
+# AWS should manage the certificate renewal automatically
+# https://docs.aws.amazon.com/acm/latest/userguide/managed-renewal.html
+# If this fails, AWS will email associated with the AWS account
+resource "aws_acm_certificate" "prometheus_cert" {
+  domain_name       = "prom.${local.subdomain}"
+  validation_method = "DNS"
+
+  subject_alternative_names = ["${aws_route53_record.prom_alias.*.fqdn}"]
+
+  lifecycle {
+    # We can't destroy a certificate that's in use, and we can't stop
+    # using it until the new one is ready.  Hence
+    # create_before_destroy here.
+    create_before_destroy = true
+  }
+}
+
+resource "aws_route53_record" "prometheus_cert_validation" {
+  # Count matches the domain_name plus each `subject_alternative_domain`
+  count = "${1 + local.prom_records_count}"
+
+  name       = "${lookup(aws_acm_certificate.prometheus_cert.domain_validation_options[count.index], "resource_record_name")}"
+  type       = "${lookup(aws_acm_certificate.prometheus_cert.domain_validation_options[count.index], "resource_record_type")}"
+  zone_id    = "${var.zone_id}"
+  records    = ["${lookup(aws_acm_certificate.prometheus_cert.domain_validation_options[count.index], "resource_record_value")}"]
+  ttl        = 60
+  depends_on = ["aws_acm_certificate.prometheus_cert"]
+}
+
+resource "aws_acm_certificate_validation" "prometheus_cert" {
+  certificate_arn         = "${aws_acm_certificate.prometheus_cert.arn}"
+  validation_record_fqdns = ["${aws_route53_record.prometheus_cert_validation.*.fqdn}"]
+}
+
+resource "aws_route53_record" "prom_alias" {
+  count = "${local.prom_records_count}"
+
+  zone_id = "${var.zone_id}"
+  name    = "prom-${count.index + 1}"
+  type    = "A"
+
+  alias {
+    name                   = "${aws_lb.prometheus_alb.dns_name}"
+    zone_id                = "${aws_lb.prometheus_alb.zone_id}"
+    evaluate_target_health = false
+  }
+}
+
+resource "aws_security_group" "prometheus_alb" {
+  name        = "${var.stack_name}-prometheus-alb"
+  description = "Access to prometheus ALB (${var.stack_name})"
+  vpc_id      = "${local.vpc_id}"
+}
+
+# Prometheus is fronted by an nginx which controls access to
+# either approved IP addresses, or users with basic auth creds
+# so we need to allow all IPs here
+resource "aws_security_group_rule" "prom_allow_http" {
```

Comment:
Optional: If I understand this correctly, it might be good to change the names of these two rules to
`prom_alb_allow_http` and `prom_alb_allow_https` so it's clear this is about allowing public traffic to the ALB but not the prometheus EC2 instance itself. The comment above this rule is also useful but could be more specific around what 'here' is (which I think is the ALB).

Reply:
Happy with this change